test/runtime/signal.c

```diff
@@ -591,6 +590,59 @@ test_sigsegv(void)
         fail_perror("siggaction for SIGSEGV failed");
 }
 
+static void sigfpe_handler(int signo) {
+  if (signo != SIGFPE)
+    fail_perror("  childtid: caught non SIGFPE signal %d\n", signo);
+
+  syscall(SYS_exit, 0);
+}
+
+static void *sigfpe_thread(void *arg) {
+  child_tid = syscall(SYS_gettid);
+  if (arg == 0) {
+
+    /* generate sigfpe */
+    asm volatile("push   %rax \t\n\
+                    xor     %rax, %rax \t\n\
+                    idiv    %rax \t\n\
+                    ");
+  } else {
+    asm volatile("hlt");
+  }
```

Comment:
No need to check for `arg == 0`, and no need for the `hlt` instruction: just execute the sigfpe-generating instructions.